src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -51,7 +51,23 @@ namespace ts.refactor.convertParamsToDestructuredObject {
         changes: textChanges.ChangeTracker,
         functionDeclaration: ValidFunctionDeclaration,
         groupedReferences: GroupedReferences): void {
-        const newParamDeclaration = map(createNewParameters(functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
+        const signature = groupedReferences.signature;
+        const newParamDeclaration = map(createNewParameters(signature ?? functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
+
+        if (signature) {
+            changes.replaceNodeRangeWithNodes(

```

Comment:
Probably should extract this and the `createNewParameters` call above into a function.